backend/homework/2/src/main/java/org/apiparser/APIResponseParser.java

```diff
@@ -0,0 +1,87 @@
+package org.apiparser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class APIResponseParser {
+    private static final Logger logger = LoggerFactory.getLogger(APIResponseParser.class);
+    public static Book parse(String response) {
+        Book book = new Book();
+
+        // Parse title
+        String endRule = "<";
+        String startRule = "<title>";
+        String title = parse(response, startRule, endRule);
+        book.setTitle(title);
+
+        // Parse author
+        startRule = "<name>";
+        String author = parse(response, startRule, endRule);
+        book.setAuthor(author);
+
+        // Parse publication year
+        startRule = "<original_publication_year type=\"integer\">";
+        int publicationYear = Integer.parseInt(parse(response, startRule, endRule));
+        book.setPublicationYear(publicationYear);
+
+        // Parse average rating
+        startRule = "<average_rating>";
+        double averageRating = Double.parseDouble(parse(response, startRule, endRule));
+        book.setAverageRating(averageRating);
+
+        // Parse ratings count
+        startRule = "<ratings_count type=\"integer\">";
+        String ratingsCountStr = parse(response, startRule, endRule).replace(",", "");
+        int ratingsCount = Integer.parseInt(ratingsCountStr);
+        book.setRatingsCount(ratingsCount);
+
+        // Parse image URL
+        startRule = "<image_url>";
+        String imageUrl = parse(response, startRule, endRule);
```

Comment:
Lets have different functions for different logic. aka Singular Responsibility(read on SOLID principles)